Swagger ui redesign/font changes

lshaw-sb opened the pull request:
This is for review only and should NOT be merged to facelift-demo

Commit:
FACELIFT Reset default text margins, Add updated spacing to information-container / scheme-container

Commit:
FACELIFT Apply UX spacing changes where reasonable and applicable

Commit:
FACELIFT Update measurement from px => em

Commit:
FACELIFT - Added missing flex alignment

taraokelly committed:
Update operation colors, abstracted colors from element specific variables

taraokelly committed:
Removed colors.scss

taraokelly committed:
Merge pull request #2 from Rody-Kirwan/swagger-redesign/update-colors

Swagger redesign/update colors

Commit:
FACELIFT - Update spacing measurements to use rem

Commit:
FACELIFT - Update spacing to rem

Commit:
FACELIFT - Set root-level font-size

Rody-Kirwan committed:
Merge pull request #1 from Rody-Kirwan/swagger-ui-redesign/reset-default-margins-padding

Reset default text margins, Add updated spacing to information-container, scheme-container

lshaw-sb committed:
first stage in updating fonts for redesign

lshaw-sb committed:
remove spaces, repace px with em

lshaw-sb committed:
refactor for em font sizes

lshaw-sb committed:
more em clean up

lshaw-sb committed:
remove margin refactor work

lshaw-sb committed:
cleanup

lshaw-sb committed:
cleanup

lshaw-sb committed:
merge conflict

lshaw-sb committed:
merge conflict

lshaw-sb committed:
more font updates

lshaw-sb committed:
more font updates

lshaw-sb committed:
clean up

lshaw-sb committed:
clean up whitespace

Rody-Kirwan commented on src/style/_fonts.scss:
```diff
@@ -0,0 +1,34 @@
+@font-face{
```
should be 4 space indentation

lshaw-sb replied:
cheers. willdo

lshaw-sb replied:
done

Rody-Kirwan commented on src/style/_information.scss:
```diff
@@ -7,32 +7,29 @@
     }
 
-    a
+    div > a, a.info__extdocs.link
```
I think this might be a bit too ambiguous - maybe add the classname to be more specific:
`div.opblock-tag > a`

lshaw-sb replied:
there are multiple a tags under the div thats why its like that

Rody-Kirwan replied:
But you are styling all direct descendant `a` elements in all `div` elements...
Adding the div class `.opblock-tag` would be a bit more specific

Rody-Kirwan commented on src/style/_information.scss:
```diff
@@ -46,21 +43,17 @@
 
     .title
     {
         font-size: 36px;
```
this should be `em` ...or removed?

Rody-Kirwan commented on src/style/_layout.scss:
```diff
@@ -207,31 +205,29 @@
     {
-        font-size: 14px;
-        font-weight: bold;
+        display: flex;
```
Is this left over from my PR?
If so you should probable delete the flex props and re-add `text-align: center`

lshaw-sb replied:
done. cheers!

Rody-Kirwan commented on src/style/_models.scss:
```diff
@@ -123,13 +123,16 @@ section.models
 
         display: flex;
         align-items: center;
+        justify-content: space-between;
```
Left over from my PR?

lshaw-sb replied:
done. cheers!

Rody-Kirwan commented on src/style/_models.scss:
```diff
@@ -123,13 +123,16 @@ section.models
         cursor: pointer;
         transition: all .2s;
 
+        border-bottom: 1px solid rgba($section-models-border-color, .3);
```
same

lshaw-sb replied:
done. cheers!

Rody-Kirwan commented on src/style/_variables.scss:
```diff
@@ -117,13 +117,17 @@ $info-title-small-background-color: $waterloo-gray !default;
 
 $info-title-small-pre-font-color: $white !default;
 
+$info-title-small-pre-font-size: 12px;
```
should this be `em`?

lshaw-sb replied:
done. cheers!

Rody-Kirwan commented on src/style/_variables.scss:
```diff
@@ -117,13 +117,17 @@ $info-title-small-background-color: $waterloo-gray !default;
 $opblock-tag-border-bottom-color: $bright-gray !default;
 $opblock-tag-background-color-hover: $black !default;
+$opblock-tag-h4-font-size: 2em;
+$opblock-tag-small-font-size:1.4em;
```
pedantic, but make sure space after colon

lshaw-sb replied:
done. cheers!

Rody-Kirwan commented:
Some tiny updates. Otherwise looks good

lshaw-sb committed:
post review changes

taraokelly commented on src/style/main.scss:
```diff
@@ -1,3 +1,8 @@
+html {
+    font-size: 62.5%;
+}
```
Is this supposed to be here?

lshaw-sb replied:
Its part of support for EM for fonts which i started to bring in from Rodys branch

Rody-Kirwan replied:
needs to be in the `.swagger-ui` block - not in `html`

Rody-Kirwan commented on src/style/_layout.scss:
```diff
@@ -57,11 +57,12 @@
-    @include text_headline();
+    a.nostyle
+    {
+      @include text_title($size: $opblock-tag-h4-font-size);
```
4 space indent

Rody-Kirwan reviewed (commented):
Just a couple of small comments. 
The main one is to move the `font-size: 62.5%` to the `.swagger-ui ` class

lshaw-sb committed:
post review changes

Rody-Kirwan commented on src/style/_type.scss:
```diff
@@ -1,21 +1,42 @@
-    font-weight: 600;
-
-    color: $color;
+  font-family: $code-default-font-family;
```
should be 4 space indentation.
I think you may need to change your settings in vscode

lshaw-sb committed:
swagger-ui selector needs greater specificity for font size

lshaw-sb committed:
updates following review from Kyle

Commit:
FACELIFT - Added environment specific font-rule

Commit:
Clean up file-loader rule

Rody-Kirwan committed:
Update index.html

Removing these changes

lshaw-sb committed:
more updates following review from Kyle

shockey committed:
generate relative public paths for font assets

avoids injection of absolute paths into production stylesheets

former output: `url(/Users/kyle/Code/ui/dist/fonts/roboto-light-webfont.woff2)`
new output: `url(./fonts/roboto-light-webfont.woff2)`

shockey committed:
add fallback font directives

makes our new `font-display: swap` play nicer; generally better for end users under poor network conditions
